src/material/core/typography/_typography.scss

```diff
@@ -8,11 +11,133 @@
 @forward './definition';
 @forward './versioning';
 
+@mixin typography-hierarchy($theme, $selector: '.mat-typography', $back-compat: false) {
+  @if inspection.get-theme-version($theme) == 1 {
+    @include _m3-typography-hierarchy($theme, $selector, $back-compat);
+  }
+  @else {
+    @include _m2-typography-hierarchy($theme, $selector);
+  }
+}
+
+@function _get-selector($selectors, $prefix) {
```

Comment:
Question: do you think there's a way to unit test to verify we don't have "naked" tag selectors, nested prefix selector, etc.?

Reply:
I can probably write a test that looks at the CSS output and verifies this

Reply:
Done